src/control/cmd/dmg/pretty/pool.go

```diff
@@ -162,15 +174,28 @@ func printListPoolsResp(out io.Writer, resp *control.ListPoolsResp) error {
 		fmt.Fprintln(out, "no pools in system")
 		return nil
 	}
+	upgrade := false
+	for _, pool := range resp.Pools {
+		if pool.HasErrors() {
+			continue
+		}
+		if pool.PoolLayoutVer != pool.UpgradeLayoutVer {
+			upgrade = true
+		}
+	}
 
-	formatter := txtfmt.NewTableFormatter("Pool", "Size", "State", "Used", "Imbalance", "Disabled")
+	titles := []string{"Pool", "Size", "State", "Used", "Imbalance", "Disabled"}
+	if upgrade {
+		titles = append(titles, "Upgrade?")
```

Comment:
the problem with introducing variable columns is that people may complain about the output changing between invocations, there seems to be an implicit expectation that the number of columns is static for a given command. we have seen this before. think it's better to keep the column list the same. we could say that if people want stability they should be using the json output but that won't please some people

Reply:
I got suggestions from @mjmac that we should only output Upgrade column if upgrade is needed...I can change this back if we get agreement on this.

Reply:
If @mjmac asked for this then keep it that way, my preference would be for static column lists but ultimately it is his decision to make.

Reply:
My reasoning on this was that the upgrade-needed state for a set of pools should be relatively transient, and therefore always including this column was unnecessary clutter for the vast majority of times that the admin will run the command to see the list of pools.